Feature/fix solution sorting

ssfinney opened the pull request:
Fixes #89.

Again, I'm having problems with comparisons on the new tests...

chipbell4 commented:
@ssfinney I think the logic's backwards. `judgeableForContest` should return sorted by created_at ascending because we want to judge the oldest solution first (out of fairness to teams). For `claimedByJudgeInContest`, we should sort *descending* by created_at because we want to see the last ones we touched sooner (since we're more likely to "oops" and change it).

chipbell4 commented:
@ssfinney Also, I patched your existing tests to pass, so you can see how I did the order comparison. However, I do think the logic is backwards.

ssfinney commented:
@chipbell4 Ok, I fixed the ordering, and added one new test (since I forgot to add an orderBy to one of the repository methods).

Thanks for fixing those tests!
